src/library.js

```diff
@@ -2078,16 +2078,11 @@ addToLibrary({
   // at runtime.
   $keepRuntimeAlive__deps: ['$runtimeKeepaliveCounter'],
   $keepRuntimeAlive: () => noExitRuntime || runtimeKeepaliveCounter > 0,
-#elif !EXIT_RUNTIME
-  // When `noExitRuntime` is not include and EXIT_RUNTIME=0 then we know the
+#elif !EXIT_RUNTIME && !PTHREADS
+  // When `noExitRuntime` is not included and EXIT_RUNTIME=0 then we know the
   // runtime can never exit (i.e. should always be kept alive).
-  // However for pthreads we always default to allowing the runtime to exit
-  // otherwise threads never exit and are not joinable.
-#if PTHREADS
-  $keepRuntimeAlive: () => !ENVIRONMENT_IS_PTHREAD,
-#else
+  // However, for pthreads we always always need to allow the runtime to exit
```

Comment:
```suggestion
  // However, for pthreads we always need to allow the runtime to exit.
```

Reply:
Maybe I was just wanting to be emphatic!  :)

Reply:
Heh, these double-positives/double-negatives always remind me of the joke,
```
A teacher told the class, "English has double negatives
that cancel out: 'I don't not want to go' means I want
to go, at least somewhat. But a double positive, unlike
in other languages, is basically the same as a positive."

A student at the back of the class scoffed, "Yeah, right."
```